src/SDKs/Azure.Core/data-plane/Azure.Core/ETag.cs

```diff
@@ -2,51 +2,42 @@
 // Licensed under the MIT License.
 
 using System;
-using System.ComponentModel;
-using System.Text;
 
 namespace Azure
 {
     public readonly struct ETag : IEquatable<ETag>
     {
-        readonly byte[] _ascii;
+        private readonly string _value;
 
-        public ETag(string etag) => _ascii = Encoding.ASCII.GetBytes(etag);
+        public ETag(string etag) => _value = etag;
+
+        public static bool operator ==(ETag left, ETag right) => left.Equals(right);
+
+        public static bool operator !=(ETag left, ETag right) => !left.Equals(right);
 
         public bool Equals(ETag other)
         {
-            if(_ascii == null) {
-                if (other._ascii == null) return true;
-                return false;
-            }
-            if (other._ascii == null) return false;
-            
-            return _ascii.AsSpan().SequenceEqual(other._ascii);
+            return string.Equals(_value, other._value);
         }
 
-        public static bool operator ==(ETag left, ETag rigth) => left.Equals(rigth);
-
-        public static bool operator !=(ETag left, ETag rigth) => !left.Equals(rigth);
+        public bool Equals(string other)
+        {
+            return string.Equals(_value, other);
+        }
 
-        public override string ToString() => _ascii == null ? "<null>" : Encoding.ASCII.GetString(_ascii);
+        public override bool Equals(object obj)
+        {
+            return obj is ETag other && Equals(other);
```

Comment:
The pattern match cast here is subtle.   I missed it at first read and thought there was a potential infinite recursion.   It may be worth wrapping in parenthesis to draw attention to the grouping and/or throwing a comment in to make it clear to casual readers.